XIPs/xip-57-disappearing-messages.md

```diff
@@ -0,0 +1,133 @@
+            disappearStartingAtNs: 1738620126404999936,
+            retentionDurationInNs: 1800000000000000
+        )
+    }
+);
+
+// Group
+await client.conversations.newGroup(
+    [address],
+    {
+        disappearingMessageSettings: DisappearingMessageSettings(
+            disappearStartingAtNs: 1738620126404999936,
+            retentionDurationInNs: 1800000000000000
+        )
+    }
+);
+```
+
+### Update disappearing message setting on an existing conversation
+
+When sending a message, it will abide by message expiration conditions set for the conversation. For example:
+
+```tsx
+await conversation.updateDisappearingMessageSettings(updatedSettings)
+await conversation.clearDisappearingMessageSettings()
+```
+
+### Get a conversation’s disappearing message setting
+
+When sending a message, it will abide by message expiration conditions set for the conversation. For example:
+
+```tsx
+conversation.disappearingMessageSettings
+conversation.isDisappearingMessagesEnabled()
+```
+
+### Automatic deletion from local storage
+
+A background worker will run every one second to clean up expired disappearing messages. The worker will automatically delete expired messages from local storage. No additional action will be required by integrators.
+
+### Automatic removal from UI
+
+Expired messages won’t require manual removal from the UI. If an app UI updates when the local storage changes, expired messages will disappear automatically when the background worker deletes them from local storage.
+
+### Receive a disappearing message
+
+On the receiving side, an app doesn't need to check expiration conditions manually. It can receive and process messages as usual, and the background worker will handle message expiration cleanup.
+
+## Rationale
+
+Some users, depending on their needs, want disappearing messages in a group or one-on-one conversation to ensure messages don’t linger longer than expected on other participants’ devices. Enabling this feature provides a way for users to enforce message expiration, knowing that as long as all clients are updated and functioning correctly, the messages will be automatically removed from those clients.
+
+## Backward compatibility
+
+Since older clients can’t process the new metadata changes, messages won’t be deleted from those clients. Also, those clients won’t recognize that disappearing messages are enabled. This means they will continue sending and displaying messages normally, but for users with updated clients, those messages will be deleted based on the expiration settings—without the sender necessarily being aware that their messages have disappeared.
+
+## Test cases
+
+See tests in [mls.rs](https://github.com/xmtp/libxmtp/blob/8771b149338fba17dd0e1fb97f6eb11bc7ba6491/bindings_ffi/src/mls.rs#L5200-L5453) in the libxmtp repo.
+
+## Reference implementation
+
+See [conversation.rs](https://github.com/xmtp/libxmtp/blob/main/bindings_node/src/conversation.rs#L698).
```

Comment:
@mchenani - does this link look correct now? <3

Reply:
Yes, Thank you! ☺️